lms/views/api/lti.py

```diff
@@ -0,0 +1,75 @@
+import datetime
+from datetime import timezone
+
+from pyramid.view import view_config
+
+from lms.validation import ApiRecordSubmissionSchema
+from lms.services.lti_outcomes import LTIOutcomesRequestParams
+
+
+@view_config(
+    request_method="POST",
+    route_name="lti_api.submissions.record",
+    renderer="json",
+    schema=ApiRecordSubmissionSchema,
+)
+def record_submission(request):
+    """
+    Record info to facilitate later grading of an assignment.
+
+    When a learner launches an assignment the LMS provides metadata that can
+    be later used to submit a score to the LMS using LTI Outcome Management
+    APIs. In Canvas, extensions to that API allow a custom LTI launch URL to be
+    submitted for use by the SpeedGrader [1].
+
+    Currently this view only supports SpeedGrader-based grading in Canvas by
+    submitting an LTI Launch URL to Canvas. In future it will need to persist the
+    metadata to facilitate grading in other LMSes via a different UI.
+
+    This work _could_ be done by the backend during an LTI launch, but as it
+    involves potentially slow requests to the external LMS, it is triggered
+    asynchronously by the frontend while the assignment is loading.
+
+    [1] https://canvas.instructure.com/doc/api/file.assignment_tools.html
+    """
+
+    lti_user = request.lti_user
+    parsed_params = request.parsed_params
+
+    shared_secret = request.find_service(name="ai_getter").shared_secret(
+        lti_user.oauth_consumer_key
+    )
+    outcome_request_params = LTIOutcomesRequestParams(
+        consumer_key=lti_user.oauth_consumer_key,
+        shared_secret=shared_secret,
+        lis_outcome_service_url=parsed_params["lis_outcome_service_url"],
+        lis_result_sourcedid=parsed_params["lis_result_sourcedid"],
+    )
```

Comment:
**Code design:** I think there's a factory function for constructing an `LTIOutcomesRequestParams` here, which could move a bunch of code out of this view. The suggestion below presents a challenge to the pattern that our value classes have no methods, so we might need to think about that (maybe the factory function belongs elsewhere).

```python
class LTIOutcomesRequestParams:
    ...

    @classmethod
    def from_request(cls, request):
        ...
        return cls(...)
```